packages/astro/src/core/routing/manifest/generator.ts

```diff
@@ -31,7 +33,7 @@ export function getRouteGenerator(
 				segment
 					.map((part) => {
 						if (part.spread) {
-							return `:${part.content.slice(3)}(.*)?`;
+							return `${part.content.slice(3)}`;
```

Comment:
This removes this part from being a parameter understood by `path-to-regexp` to just being a literal, e.g. if the route was `[...slug]`, previously this template literal created `:slug(.*)` but after this change it just creates `slug`. That’s probably why a lot of tests are failing.

We’ll need to check how to match the `(.*)` behaviour in the latest version of `path-to-regexp` (if possible).

Reply:
From taking a look at the docs, it seems like this might be the way?

```suggestion
							return `*${part.content.slice(3)}`;
```

Reply:
I tried this as well. It errors out and says "expected "slug" to be a non-empty array"

Reply:
Ah, I guess currently we allow `[...spread]` to match nothing (i.e. `[...spread]` can match `/`) and `*spread` in `path-to-regexp` doesn’t include support for an optional segment. They do mention `{}` for [optional segments](https://github.com/pillarjs/path-to-regexp#optional), could try combining that with the wildcard?

Reply:
Then there is still the issue with the actual parameter value for rest (...) parameters, today it's "somedir/file.ext" and with the new version it should be ["somedir", "file.ext"], i.e. the parameter value needs to be split.

Reply:
Okay. I made some changes and im testing it all now to see.